Ex10_Answers.py

```diff
@@ -0,0 +1,159 @@
+store_N0s=pandas.DataFrame({"time":times,"N01":0,"N02":0,"N03":0}) #Initialize a data frame to store initial sizes with columns for each
+                                                               #initial size, and the times modeled
+
+for i in range(0,len(N0s)): #Set up a for loop that runs through each N0 in the list and models population growth using
+
+    params=(r,K)#other established parameters
+    N0=N0s[i] #Assign N0 a single value from the array of initial population sizes
+    sim=spint.odeint(func=ddSim,y0=N0,t=times,args=params)
+    store_N0s.iloc[:,i]=sim[:,0] #Store population values in the column appropriate for the current carrying capacity
+
+initpop=ggplot(store_N0s,aes(x="time",y="N"))+theme_classic()#Set up base plot for population curves
+initpop=initpop+geom_line(store_N0s,aes(y="N01"))
+initpop=initpop+geom_line(store_N0s,aes(y="N02"))
+initpop=initpop+geom_line(store_N0s,aes(y="N03"))
+
+initpop #Display graph
+
+# Question 2
+
+#########################################################
+
+import scipy
+import scipy.integrate as spint
+import numpy
+import matplotlib.pyplot as plt
+
+def SIR_model(y, t, beta, gamma):
+    S = y[0]
+    I = y[1]
+    R = y[2]
+    
+    dSdt=(-beta*I*S)
+    dIdt=(beta*I*S-(gamma*I))
+    dRdt=(gamma*I)
+    
+    return (dSdt, dIdt, dRdt)
+
+N = [999, 1, 0]
+S0=999
+I0=1
+R0=0
+beta=numpy.array([.0005, .005, .0001, .00005, .0001, .0002, .0001])
+gamma=numpy.array([.05, .5, .1, .1, .05, .05, .06])
+
+store_S=pandas.DataFrame({"time":times,"beta1":0,"beta2":0,"beta3":0,"beta4":0,"beta5":0,"beta6":0,"beta7":0})
+store_I=pandas.DataFrame({"time":times,"beta1":0,"beta2":0,"beta3":0,"beta4":0,"beta5":0,"beta6":0,"beta7":0})
+store_R=pandas.DataFrame({"time":times,"beta1":0,"beta2":0,"beta3":0,"beta4":0,"beta5":0,"beta6":0,"beta7":0})
+
+t=numpy.linspace(0,100)
+
+for i in range(0,len(beta)):
+    params=(beta[i],gamma[i])
+    sim=spint.odeint(func=SIR_model,y0=N,t=times,args=params)
+    store_S.iloc[:,i]=sim[:,0]
+    store_I.iloc[:,i]=sim[:,1]
+    store_R.iloc[:,i]=sim[:,2]
+
+
+plt.figure(figsize=[6,4])
+plt.plot(t, solution[:, 0], label="S(t)")
+plt.plot(t, solution[:, 1], label="I(t)")
+plt.plot(t, solution[:, 2], label="R(t)")
+plt.show()
+
+#max daily incidence
+mdi=np.diff(store_I.iloc[:,1])
+Imax=numpy.max(mdi)
+print("max daily incidence")
+print(Imax)
+
+#max daily prevelance
+mdp=Imax/1000
+print("max daily prevalence")
+print(mdp)
+
+#percent affected over simulation
+smallS=numpy.min(store_S.iloc[:,0])
+pas=(1/smallS)
+print("percent affected over simulation")
+print(pas)
+
+#basic reproduction number
+S0=999
+I0=1
+R0=0
+beta=numpy.array([.0005, .005, .0001, .00005, .0001, .0002, .0001])
+gamma=numpy.array([.05, .5, .1, .1, .05, .05, .06])
+basicReproductionNumber=(((beta*(S0+I0+R0))/gamma))
+print("BasicReproductionNumber")
+print(basicReproductionNumber)
+
+print("!!! please see file named README for analysis completing question 2 !!!")
```

Comment:
storeSIR=pandas.DataFrame({"beta":beta,"gamma":gamma,"R0":beta*sum(y0)/gamma,"maxIncidence":0,"maxPrevalence":0,"percentAffected":0}) .  # make the data frame to store the three calculations

for i in range(0,len(betas)):
    pars=(beta[i],gamma[i])    # loop through betas and gammas
    
    sim=si.odeint(func=simSIR,y0=y0,t=times,args=pars) . # simulates SIRs
    
    storeSIR.iloc[i,4]=numpy.max(sim[:,1]/numpy.sum(sim,axis=1))   # calculates the max prevalence
    storeSIR.iloc[i,3]=numpy.max(sim[1:len(sim),1]-sim[0:(len(sim)-1),1]) # calculates the max incidence
    storeSIR.iloc[i,5]=numpy.sum(sim[len(sim)-1,1:3])/numpy.sum(sim[len(sim)-1,:])*100 # calculates the max percentat affected